src/platform/ESP32/ESP32FactoryDataProvider.cpp

```diff
@@ -230,6 +230,93 @@ CHIP_ERROR ESP32FactoryDataProvider::GetRotatingDeviceIdUniqueId(MutableByteSpan
 #endif
     return err;
 }
+
+CHIP_ERROR ESP32FactoryDataProvider::GetSerialNumber(char * buf, size_t bufSize)
+{
+    ChipError err       = CHIP_NO_ERROR;
```

Comment:
Rather than duplicating code you can call the base class method here?
`GenericDeviceInstqnceInfoProvider::GerSerialNumber()`